src/data-access/LanguageItemSearcher.ts

```diff
@@ -0,0 +1,24 @@
+import ItemSearcher, { SearchedItemOption } from '@/data-access/ItemSearcher';
+import MwApiItemSearcher from '@/data-access/MwApiItemSearcher';
+
+export default class LanguageItemSearcher implements ItemSearcher {
+
+	private readonly LANGUAGE_PROFILE_NAME: string = 'language' as const;
+
+	private itemSearcher: MwApiItemSearcher;
+	private availableSearchProfiles: string[];
+
+	public constructor( itemSearcher: MwApiItemSearcher, availableSearchProfiles: string[] ) {
+		this.itemSearcher = itemSearcher;
+		this.availableSearchProfiles = availableSearchProfiles;
+	}
+
+	public async searchItems( searchTerm: string, offset?: number ): Promise<SearchedItemOption[]> {
+		const additionalParams: Record<string, string> = {};
+		if ( this.availableSearchProfiles.includes( this.LANGUAGE_PROFILE_NAME ) ) {
+			additionalParams.profile = this.LANGUAGE_PROFILE_NAME;
```

Comment:
We could also move this into the constructor, right? (And then maybe even discard `this.availableSearchProfiles`.)

Reply:
Not sure if I fully understand what you are suggesting. I adjusted it a bit. Let me know what you think

Reply:
What I had in mind was that even the `additionalParams` could be a class member, but I’m also okay with just having the `useLanguageProfile` boolean there.